app/models/post.rb

```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+    belongs_to :author, class_name: 'User'
+    has_many :comments
+    has_many :likes
+    def update_posts_counter
+        user.increment!(:posts_counter)
+      end
```

Comment:
- [Optional] You can equally use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the update_posts_counter method every time a new post has been saved to the database, then make the update_post_counter method private.

This can equally be done for the update_comments_counter and update_likes_counter methods in their various controller.